### ISSUE TITLE
ValueSet.TryAdd in IndexProviderTransformingIndexValues doesn't exist after upgrade

### ISSUE THREAD

**Issue by wadrschall** [Shazwazza/Examine#285]
After upgrading our Umbraco from V9 to V10, the following no longer works:

![image](https://user-images.githubusercontent.com/50219294/176771165-0db87f78-9787-44ac-8490-c4bc9d1cc695.png)

The specific error is that ValueSet does not contain a definition for TryAdd(). 

Based on this documentation https://our.umbraco.com/documentation/reference/searching/examine/examine-events, it appears it should still work. Can you let us know if this is an error/oversight or if there is an alternative way to add these custom fields in the newer version of examine?

Much appreciated!

**Comment by wadrschall** [comment]
I did find that this workaround will work. Its not how the documentations shows it and its a little messier to read, but if I clone the Dictionary, add my 2 items and then set the EventArgs object to use that new dictionary, it works.

![image](https://user-images.githubusercontent.com/50219294/176776766-dbca0d6d-6a53-4e5b-bf0f-c8a9605a1794.png)

**Comment by Shazwazza** [comment]
Yes ValueSet is immutable now. You can read all about it here https://github.com/Shazwazza/Examine/issues/276

**Comment by Shazwazza** [comment]
Duplicates #281

**Comment by wadrschall** [comment]
Do you know if the Umbraco documentation can be updated to reflect these changes? I was able to figure out that workaround, but it would be helpful for the community to know that's the official way to do it.

**Comment by Shazwazza** [comment]
Yep there's an "Edit this page" button at the top of that docs page https://our.umbraco.com/documentation/reference/searching/examine/examine-events
